.github/workflows/build-and-deploy.yml

```diff
@@ -0,0 +1,168 @@
+name: Continuous deployment
+on:
+  push:
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    env:
+      deploy_staging: ${{ github.event_name == 'pull_request' || (github.event_name == 'push' && (github.ref == 'refs/heads/main' || github.ref == 'refs/heads/master')) }}
+    outputs:
+      deploy_staging: ${{ env.deploy_staging }}
+      
+    defaults:
+      run:
+        shell: bash
+        working-directory: ./src
+    
+    steps:       
+      - name: Checkout 🛎️
+        uses: actions/checkout@v2 # If you're using actions/checkout@v2 you must set persist-credentials to false in most cases for the deployment to work correctly.
+        with:
+          persist-credentials: false
+
+      - name: Cache 💾
+        uses: actions/cache@v2
+        with:
+          path: ${{ github.workspace }}/.next/cache
+          key: ${{ runner.os }}-nextjs-${{ hashFiles('**/package-lock.json') }}
+
+      - name: Setup node 🔧 # This example project is built using npm and outputs the result to the 'build' folder. Replace with the commands required to build your project, or remove this step entirely if your site is pre-built.
+        uses: actions/setup-node@v2
+        with:
+          node-version: '14'
+        env:
+            CI: true
+
+      - name: Export static site 💻
+        run: |
+          npm install
+          npm run export
+          cp -r .circleci _static
+          touch _static/.nojekyll
+        working-directory: ${{ github.workspace }}
+       
+      - name: Cache build artifacts 📦
+        if: ${{env.deploy_staging == 'true'}}
+        uses: actions/cache@v2
+        with:
+          path: ${{ github.workspace }}/_static
+          key: ${{ runner.os }}-static-site-${{ github.sha }}
+  
+  stage:
+    runs-on: ubuntu-latest
+    needs: build
+    env:
+      heroku_api_key: ${{ secrets.HEROKU_API_KEY }}
+      heroku_email: ${{ secrets.HEROKU_EMAIL }}
+      heroku_app: ${{ secrets.HEROKU_STAGING_APP }}
+    if: needs.build.outputs.deploy_staging == 'true'
+    
+    defaults:
+      run:
+        shell: bash
+        working-directory: ./src
+    
+    steps:
+    
+      - name: Checkout 🛎️
+        uses: actions/checkout@v2 # If you're using actions/checkout@v2 you must set persist-credentials to false in most cases for the deployment to work correctly.
+        with:
+          persist-credentials: false
+          
+      - name: Setup heroku CLI 🛠
```

Comment:
s/Setup heroku CLI/Setup Heroku CLI/

Reference: [Heroku is capitalized or not?](https://www.google.com/search?q=heroku+is+capitalized+or+not&rlz=1C1SQJL_itIT952IT952&sxsrf=ALeKk00-PAUNYHvZpI_fxumo8zz4V2DpVw%3A1622613777652&ei=ER-3YOemJ86I9u8P-Zi9iAg&oq=heroku+is+capitalized+or+not&gs_lcp=Cgdnd3Mtd2l6EAM6BwgAEEcQsAM6BQghEKABOggIIRAWEB0QHlC-KljjMGCTNmgBcAJ4AIABc4gBvAWSAQM3LjGYAQCgAQGqAQdnd3Mtd2l6yAEIwAEB&sclient=gws-wiz&ved=0ahUKEwin88uuo_jwAhVOhP0HHXlMD4EQ4dUDCA8&uact=5)